core/domain/topic_domain.py

```diff
@@ -54,6 +54,7 @@
 TOPIC_PROPERTY_URL_FRAGMENT = 'url_fragment'
 TOPIC_PROPERTY_META_TAG_CONTENT = 'meta_tag_content'
 TOPIC_PROPERTY_PRACTICE_TAB_IS_DISPLAYED = 'practice_tab_is_displayed'
+TOPIC_PROPERTY_PAGE_TITLE = 'page_title'
```

Comment:
Oh, can't we just use the name here, why a new field?
Also, what about the cards shown in the classroom page, are they topic names or page titles?

Reply:
Page title will be a custom text shown in the browser tab in the topic viewer page, it need not be the same as the topic name. For example, for fractions, the planned page title will be `Learn Fractions | Add, Subtract, Multiply and Divide | Oppia` where "Fractions" is taken from the name of the topic dynamically and the middle part "Add, Subtract, Multiply and Divide" will be provided by this field.

Reply:
Perhaps conceptually it may make more sense to frame that extra part as "additional information" or similar related to the topic, rather than creating what seems to be a UI-only field as part of the domain layer. Do you have a few other examples to list here to help provide more context? That might make it easier to figure out a good name for the field that contains this value.

Reply:
Some background: For SEO purposes, we have 2 fields, one is the meta tag content (for Topics and Stories) and the other is the page title field (introduced in this PR for Topics). We explain what these fields are for in the UI using help text below the input fields so that the creator is aware what these input fields represent and enter stuff accordingly.

Regarding naming: I felt, for a dev, it is immediately apparent what the field represents by its name, so I didn't feel the need to call this by a different name. Also, we won't really be using this for any other purpose other than the page title in the viewer page. That said, considering that it represents a part of the page title and not the exact page title, I am open to renaming it if we can find a better name for it that accurately represents what it is. Maybe "page title fragment" like url fragment -- not sure?

Also, I don't have other examples, maybe @seanlip or @dchen97 can share some examples. @dchen97 / @seanlip can you please give us a few examples of what the page titles for topics other than Fractions would look like? Thanks!

Reply:
For some additional context on my thinking: it's generally an antipattern to reference UI-specific data in the domain layer of an application since it introduces an awkward dependency (consider that the domain layer of an application can be displayed in different situations than the one immediately obvious). When considering a platform like Android, the domain layer should (for the most part) be the same for both web & Android frontends, but 'page title' is very clearly web-based and breaks that level of separation.

Reply:
Deferring to @dchen97 to give examples of actual titles.

@BenHenning: if we don't put this info here, then where do we put it? I think it is entirely reasonable to keep it where it is at the moment since it is part of the topic. Other clients that make use of the data can simply ignore fields which are not relevant, or the client can have an adapter layer which changes the response from the backend API into one that only contains fields needed by that client before passing it to the rest of the client application. I think this is good general practice (the same considerations apply to the web frontend).

With regards to naming, I don't think the name of the field needs to be changed, but feel free to make suggestions if you have a better name. I am reluctant to make such a change anyway because this would mean that existing fields like meta_tag_content would also need to be changed for similar reasons and I don't see the upside.

Reply:
To be clear, I am not suggesting we don't store the data.

My concern here is that the naming will lead to potential confusion. It seems to be a violation of the contract of the domain layer to care about the web frontend--that's what the web controllers are for. Per single responsibility, my understanding of the the responsibility of the domain layer of the Oppia stack is to maintain the model & business logic that is needed to perform the core functions of Oppia (which to me is supporting playing lessons & related content, and creating that content). That purpose is clearly independent of how clients decide to render the data.

The reason I'm asking for more examples is because I suspect that the data concept that we want to store here is actually generalizable and can be framed as not specific to an individual client. Meta tags may also apply--I'm not sure seeing as I also don't fully understand the use cases there yet. However, until I have more information, one potential interpretation of this page title might be: if the secondary page title is meant to enumerate key skills that the learner may learn, then storing them as "key_skills_to_learn" or "skills_to_advertise_to_learner" or something along those lines would no longer tie the concept to the web frontend. While Android itself may not use this out of the box, the idea that it conveys could surely be useful on Android in the future because of its domain applications rather than its web applications.

I recognize that we have lots of existing examples of web-specific fields stored in the data store, and in some regards that's fine since Oppia has been, until now, a web-based platform. As we build our second major frontend and consider the implications of generalizing our backend to multiple frontends, I think framing the domain layer in terms of the functional part of the product rather than how or where we display that to the user will result in a more maintainable codebase over time. This may require renaming old fields to something that's more generalizable.

I also recognize that this is largely a reflection of my own interpretation of single responsibility & generalizing codebases, and am more than willing to continue discussing this point in away that doesn't block this PR. :) I evidently have fairly strong opinions about this, and while I can make arguments about how I've seen similar cases in the past lead to developer confusion, there are likely just as many arguments in favor of "this probably won't actually result in any problems" and that very well may be true in this case. I tend to be more of a "purist" when it comes to architecture, and to that end I'm happy to write up some general API development guidelines that we can discuss in a future core maintainers meeting. This PR doesn't need to block on that, but any input folks have here will help influence how I would frame such guidelines. I will also need to evaluate how existing documentation may need to change, as well.

Reply:
Tagging:
- @dchen97 to confirm format of titles
- @BenHenning to confirm residual changes (we had a discussion and I think the two main things are: (a) change the field name to page_title_for_web, and (b) have the field store only the variable bits of the title -- this is dependent on @dchen97's response above).

Reply:
Hey all - I don't have a clear idea of what is the best way to standardize titles, but below are some examples (though know note a full list is available https://docs.google.com/spreadsheets/d/16COU_tcKnB0ygBAy6GyJlexVRoP3FzN5VvAh_gZRqV4/edit#gid=0):

For topic pages: "Learn [TOPIC - e.g. Fractions] | [Set of keywords - in case of fractions "Add, Subtract, Multiply, and Divide" | Oppia"
For Math Classroom: "Learn Math with Oppia | Oppia"
For About page: "About Oppia | Oppia"
For Community Library: "Community Library | Learn from over 20,000 Lessons | Oppia"

Generally, unless the title is very self-explanatory or it's a more simplistic web page, we should have 3 sections separated by "|" dividers with the last section ALWAYS saying Oppia. For more simple web pages, we should just have 2 pages with the first being a concise description of the page followed by "Oppia."

Reply:
Thanks all! I've used "page_title_for_web" for now.

Done.

Reply:
Not done, needs one more commit to change the name to "page_title_fragment_for_web" since the field represents a fragment of the page title.

Reply:
Sorry for not responding earlier--I dropped this. Thanks @seanlip for summarizing our discussion.

I am interested in the idea that we're going to have page titles for more than just topics. This doesn't quite reflect that need. Also, @dchen97 was suggesting using keywords for topics--it would be cleaner to model this as ``keywords_for_page_title_web``, or, do we want to have the flexibility for non-keyword titles?

Regardless, I'm happier with the new name. :)

Reply:
Also to clarify one point: should we be modeling page titles separately if each page is going to have a customizable title? Or, are we going to manually change the static pages & only allow this level of customization for topics?

Reply:
Let's keep this PR scoped to topics. The other examples @dchen97 mentioned are for static pages whose titles are defined at the platform layer and not by the creator. (I think this clarifies your second point, @BenHenning.) 

Also, I would prefer to keep the flexibility here with a string representation. It's possible that, going forward, not all our topic titles will be in keyword form, so I don't want to hardcode that assumption here.

Reply:
Ack, will take a follow-up review today. Thanks!